nomad/structs/structs.go

```diff
@@ -11428,6 +11443,59 @@ func (a *Allocation) NeedsToReconnect() bool {
 	return disconnected
 }
 
+// GetLeaderTask will return the leader task in the allocation
+// if there is one, otherwise it will return an empty task.
+func (a *Allocation) GetLeaderTask() Task {
```

Comment:
By returning `Task` instead of `*Task` we end up doing a shallow-copy of the struct. Given that the caller is going to need to detect the "empty task" case anyways, maybe we should return a `*Task` here and let the caller grab the first task if it's nil?

Reply:
I don't like returning pointers unless nil has a meaning, because you leave to the caller the task of checking for nil before using the result